### ISSUE TITLE
Allow "Developer tools" to write to the global clipboard

### ISSUE THREAD

**Issue by lisa-wolfgang** [ScratchAddons/ScratchAddons#1004]
**Is your feature request related to a problem? Please describe.**
It would be nice to be able to use the "Copy All" and "Copy Block" functions in the "Developer tools" addon across different projects as a cleaner method than the backpack.

**Describe the solution you'd like**
When using these functions, instead of writing to webpage-restricted storage, the addon would write to the user's clipboard, similar to how the "Bitmap images copying" addon works. It could then be pasted on other projects that the user has open or even in a text file for later usage.

**Describe alternatives you've considered**
Using the backpack/exporting... 😐
They're just so slow and clunky.

**Comment by areeq0325** [comment]
Maybe blocks turn to [scratchblocks](https://scratchblocks.github.io)?

**Comment by Explosion-Scratch** [comment]
Blocks turning into scratchblocks would be great! That and then you could also paste scratchblocks!!!

**Comment by apple502j** [comment]
Ok, maybe time to add parse-sb3-blocks into the list of dependencies.

**Comment by jeffalo** [comment]
ive spoken to griffpatch man about this and he's said that it would be a bit jarring if your clipboard unexpectedly became scratch blocks.

**Comment by WorldLanguages** [comment]
No need to actually fill the clipboard. We can use local storage, and if there's both a block copied in this project, and from another project, we could show "Paste block from this project" and "Paste block from external project"

**Comment by lisa-wolfgang** [comment]
> ive spoken to griffpatch man about this and he's said that it would be a bit jarring if your clipboard unexpectedly became scratch blocks.
> 
> perhaps there's a way to store it but not as text?
> 
> what about cookies or local storage

Perhaps this could be a checkbox option in the Developer tools addon that’s disabled by default. That way users would know what they’re getting into.

**Comment by Touchcreator** [comment]
No lol

**Comment by TheColaber** [comment]
> No lol

Why not??

**Comment by Touchcreator** [comment]
> 
> 
> > No lol
> 
> Why not??

What if you press crtl + v to paste a number, but instead you pasted scratch blocks

**Comment by TheColaber** [comment]
> > > No lol
> > 
> > 
> > Why not??
> 
> What if you press crtl + v to paste a number, but instead you pasted scratch blocks

That's not the issue... we wouldn't change anything from the current system besides that if you press Ctrl+C it would save it to the local storage... I don't see an issue...

**Comment by Touchcreator** [comment]
> 
> 
> > > > No lol
> > > 
> > > 
> > > Why not??
> > 
> > 
> > What if you press crtl + v to paste a number, but instead you pasted scratch blocks
> 
> That's not the issue... we wouldn't change anything from the current system (of pasting) besides that if you press Ctrl+C/right click + copy it would save it to the local storage... I don't see an issue...

I guess it's fine

**Comment by Cool-Scratcher** [comment]
> ive spoken to griffpatch man about this and he's said that it would be a bit jarring if your clipboard unexpectedly became scratch blocks.
> 
> perhaps there's a way to store it but not as text?
> 
> what about cookies or local storage

I agree with griffpatch. It would be a bit random if I suddenly pasted Scratchblocks. However, it would also be strange if I pasted Scratchblocks. Like @WorldLanguages said, using local storage would work well.

Maybe when you copy a block or something, you can't paste it anywhere else, because it's not possible. However, when you paste it into another Scratch editor, it pastes _as blocks_.
Using a special key combination (Ctrl+Shift_V is paste raw text with no styling in Google Docs, maybe that) you can paste _Scratchblocks code_.
Just my 2 cents.
